src/Axiom/Set/Map.agda

```diff
@@ -184,6 +195,23 @@ m ∣' P? = filterᵐ (sp-∘ P? proj₁) m
 _∣^'_ : {P : B → Type} → Map A B → specProperty P → Map A B
 m ∣^' P? = filterᵐ (sp-∘ P? proj₂) m
 
+mapPartialLiftKey-just-uniq : ∀ {a : A} {b b' : B'} {f : A → B → Maybe B'} {r : Rel A B}
+  → left-unique r
+  → just (a , b) ∈ mapˢ (mapPartialLiftKey f) r
+  → just (a , b') ∈ mapˢ (mapPartialLiftKey f) r
+  → b ≡ b'
+mapPartialLiftKey-just-uniq {f = f} prop a∈ a'∈ with mapPartialLiftKey-map {f = f} a∈ | mapPartialLiftKey-map {f = f} a'∈
+... | _ , eq , ax∈r | _ , eq' , ax'∈r with prop ax∈r ax'∈r
+... | refl with trans eq (sym eq')
+... | refl = refl
+
+mapPartial-uniq : ∀ {r : Rel A B} {f : A → B → Maybe B' } → left-unique r → left-unique (mapPartial (mapPartialLiftKey f) r)
+mapPartial-uniq {f = f} prop {a} {b} {b'} p q with to ∈-map ((a , b) , refl , p) | to ∈-map ((a , b') , refl , q)
```

Comment:
```suggestion
mapPartial-uniq {f = f} prop p q with to ∈-map ((_ , refl , p) | to ∈-map (_ , refl , q)
```
Maybe we should have a theorem that does `to ∈-map` with `refl` (i.e. of type `a ∈ X → f a ∈ map f X`)